### ISSUE TITLE
Add Learning Rate Decay

### ISSUE THREAD

**Issue by leonardoaraujosantos** [leonardoaraujosantos/DLMatFramework#14]
## Introduction
Allows the learning rate to decay automatically while the we're training the network.

### Types of decay
* Step Decay: Reduce the learning rate by some factor every few epochs. Typical values might be reducing the learning rate by a half every 5 epochs, or by 0.1 every 20 epochs.
* Exponential Decay: 
* 1/t Decay: 

In practice, we find that the step decay dropout is slightly preferable because the hyperparameters it involves (the fraction of decay and the step timings in units of epochs) are more interpretable than the hyperparameter kk. Lastly, if you can afford the computational budget, err on the side of slower decay and train for a longer time.

## Reference
* [Cs231n Notes](http://cs231n.github.io/neural-networks-3/#anneal)
* [Cs231n Solver class](https://github.com/leonardoaraujosantos/DLMatFramework/blob/master/learn/cs231n/assignment2/cs231n/solver.py)
* [Cs231n(old) Classifer class](https://github.com/huyouare/CS231n/blob/master/assignment2/cs231n/classifier_trainer.py)
